src/main/java/io/r2dbc/pool/ConnectionPool.java

```diff
@@ -45,37 +44,27 @@
     /**
      * Creates a new connection factory.
      *
-     * @param configuration the configuration to use connections
+     * @param configuration the configuration to use for building the connection pool.
      * @throws IllegalArgumentException if {@code configuration} is {@code null}
+     * @throws IllegalArgumentException if {@link ConnectionPoolConfiguration#getConnectionFactory()} is {@code null}
+     * @throws IllegalArgumentException if {@link ConnectionPoolConfiguration#getInitialCustomizer()} is {@code null}
+     * @throws IllegalArgumentException if {@link ConnectionPoolConfiguration#getCustomizer()} is {@code null}
+     * @throws IllegalArgumentException if {@link ConnectionPoolConfiguration#getMaxCreateConnectionTime()} is {@code null}
+     * @throws IllegalArgumentException if {@link ConnectionPoolConfiguration#getMaxAcquireTime()} is {@code null}
      */
     public ConnectionPool(ConnectionPoolConfiguration configuration) {
-        this(Assert.requireNonNull(configuration.getConnectionFactory(), "ConnectionPoolConfiguration must not be null"),
-                Assert.requireNonNull(configuration.getMaxCreateConnectionTime(), "MaxCreateConnectionTime must not be null"),
-                Assert.requireNonNull(configuration.getMaxAcquireTime(), "MaxAcquireTime must not be null"),
-                configuration.getCustomizer());
-    }
+        Assert.requireNonNull(configuration, "ConnectionPoolConfiguration must not be null");
+        Assert.requireNonNull(configuration.getConnectionFactory(), "ConnectionFactory must not be null");
```

Comment:
We expect a `ConnectionPoolConfiguration` to be valid, this means that the object can be either valid or `null` (where we throw `IllegalArgumentException`). A valid object never returns `null` for `getConnectionFactory()` and such.